README.rst

```diff
@@ -542,14 +542,42 @@ List topologies
 
 .. code-block:: text
 
-    GET /api/v1/topology/
+    GET /api/v1/topology/topology/
+
+Create topology
+###############
+
+.. code-block:: text
+
+    POST /api/v1/topology/topology/
```

Comment:
> receive strategy's receive Url is not visible in API, check the UI:
Yes @atb00ker this is handled by frontend

Reply:
> 3. I can put Url value for receive strategy and it doesn't complain.

Yes, it is implemented in this way in the admin panel.

Reply:
> 2. we should create a new key (in) if the key is not send in the body of the create request or we should give an error saying "please send a key". Since the key autogenerated in the UI, I think we should generate and send back the key, what do you think? @nemesisdesign @ManishShah120 @okraits .

Hi, This is how it is being implemented in the admin panel, i.e, if the strategy is `fetch`, with no key the topology can be created,(if the user wantes he can clear the auto generated key). So should I introduce a method, like if the field is empty , then key is auto generated, for both fetch and receive strategy.

Reply:
> Yes, it is implemented in this way in the admin panel.

No, in the UI, the URL is made to be read-only in receive strategy, please check it! :smile:
You can't enter value for it! :smile:

Reply:
> Hi, This is how it is being implemented in the admin panel, i.e, if the strategy is fetch, with no key the topology can be created,(if the user wantes he can clear the auto generated key). So should I introduce a method, like if the field is empty , then key is auto generated, for both fetch and receive strategy.

No, you can't give it `key` value in Fetch mode, in the UI, input is hidden in API, it should complain if I forcefully try to provide it a value! :smile:

Reply:
> No, you can't give it key value in Fetch mode, in the UI, input is hidden in API, it should complain if I forcefully try to provide it a value! smile

The key is always present and saved.


@atb00ker there are only one `URL` field present in the topology model, which is only used when the type is set to  `fetch`, the one which is shown when the type is `receive` doesn't come from the database.

See this:-
![to_Show](https://user-images.githubusercontent.com/45564404/129450704-25dc8ce1-5333-48d9-987b-c0b047c7add9.gif)

Exception is raised only in two case, 
 - When `Key` is not provided with receive strategy.
 - when `url` is not provided with fetch strategy.
(These two are handled in the API)

Reply:
Okay, got it.
A little odd behavior but it's not a hard blocker for sure! :smile: